File: 0448-find-all-numbers-disappeared-in-an-array/solution.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        boolean[] s=new boolean[nums.length+1];
        List<Integer> m=new ArrayList<>();
        for(int num:nums){
            s[num]=true;
        }
        for(int i=1;i<=nums.length;i++){
            if(!s[i]){
                m.add(i);
            }
        }
        return m;
    }
}

File: 0448-find-all-numbers-disappeared-in-an-array/solution.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        i=0
        n=len(nums)

        while i<n:
            c=nums[i]-1
            if nums[i]!=nums[c]:
                nums[i],nums[c]=nums[c],nums[i]
            else:
                i+=1
        
        res=[]
        for i in range(n):
            if nums[i]!=i+1:
                res.append(i+1)
        return res
